src/object.c

```diff
@@ -564,7 +562,7 @@ void decrRefCount(robj *o) {
             case OBJ_HASH: freeHashObject(o); break;
             case OBJ_MODULE: freeModuleObject(o); break;
             case OBJ_STREAM: freeStreamObject(o); break;
-            default: serverPanic("Unknown object type"); break;
+            default: serverPanic("Unknown object type");
```

Comment:
This is the kind of changes that can introduce some compiler warnings in some unusual setups, like some old GCC version, etc. The change is unnecessary and not worth the risk IMHO.